app/src/main/java/org/oppia/app/player/audio/CellularDataDialogFragment.kt

```diff
@@ -0,0 +1,59 @@
+package org.oppia.app.player.audio
+
+import android.app.Dialog
+import android.content.Context
+import android.os.Bundle
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
+import android.widget.CheckBox
+import org.oppia.app.R
+
+/**
+ * DialogFragment that indicates to the user they are on cellular when trying to play an audio voiceover.
+ */
+class CellularDataDialogFragment : DialogFragment() {
+  private var doNotShowAgain: Boolean = false
+
+  companion object {
+    lateinit var cellularDataInterface: CellularDataInterface
+    /**
+     * This function is responsible for displaying content in DialogFragment.
+     * @param cellularDataInterface: [CellularDataInterface] to send data back to parent
+     * @return [CellularDataDialogFragment]: DialogFragment
+     */
+    fun newInstance(
+      cellularDataInterface: CellularDataInterface
+    ): CellularDataDialogFragment {
+      this.cellularDataInterface = cellularDataInterface
+      return CellularDataDialogFragment()
+    }
+  }
+
+  override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+    val view = activity!!.layoutInflater.inflate(R.layout.cellular_data_dialog, null)
+    val checkBox = view.findViewById<CheckBox>(R.id.cellular_data_dialog_checkbox)
+
+    checkBox.setOnCheckedChangeListener { buttonView, isChecked ->
+      doNotShowAgain = isChecked
+    }
+
+    return AlertDialog.Builder(activity as Context)
+      .setTitle(R.string.cellular_data_alert_dialog_title)
+      .setView(view)
+      .setMessage(R.string.cellular_data_alert_dialog_description)
+      .setPositiveButton(R.string.cellular_data_alert_dialog_okay_button) { dialog, whichButton ->
+        if (cellularDataInterface != null) {
+          cellularDataInterface.enableAudioWhileOnCellular(doNotShowAgain)
+        }
+        dismiss()
+      }
+      .setNegativeButton(R.string.cellular_data_alert_dialog_cancel_button) { dialog, whichButton ->
+        if (cellularDataInterface != null) {
+          cellularDataInterface.disableAudioWhileOnCellular(doNotShowAgain)
+        }
+        dismiss()
+      }
+      .setCancelable(true)
```

Comment:
What will cancellation do? Close the dialog without enabling/disabling audio?

Reply:
If the dialog has been shown, and now if the user clicks on back-button from android phone and not the application, then the dialog will close automatically without doing  anything.

If we set this as false, user will get stuck to either select "OKAY" or "CANCEL" in the dialog and default android-back button won't work.

Note: By default-android-back-button I mean back button in navigation that android OS have.

Reply:
Ah, I didn't know the nuance of setting it after creating, but my question was actually whether cancellation does the right thing here from the user's perspective. I know it automatically dismisses the dialog, but what will the user's experience be if they see this dialog and cancel it rather than dismiss/press okay?

Reply:
even if we use `setCancelable(true or false)` everything works fine because it does not work here in this case. That's why I have removed this. and the default behaviour of android on dialogs works correct for our case, which is, if the dialog is shown and user clicks on back button, the dialog will dismiss itself.